ui/components/TracePage/PromptEditor.tsx

```diff
@@ -0,0 +1,379 @@
+    const res = await fetch(`${backendUrl}/agents/${task}`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-Api-Key": process.env.NEXT_PUBLIC_BACKEND_API_KEY ?? "",
+      },
+      body: JSON.stringify({
+        agent: selectedAgent,
+        prompt: currentPrompt,
+        options: task === "classify" ? classifyOptions : undefined,
+        stop: multiline ? stop : [...stop, "\n"],
+      }),
+    });
+    const data = await res.json();
+    if (task === "classify") setClassifyResults(data);
+    if (task === "complete") setPromptResult(data);
+    setLoading(false);
+  };
+
+  return (
+    <div
+      className={`${
+        open ? "flex" : "hidden"
+      } h-screen w-screen absolute top-0 left-0 bg-gray-200 bg-opacity-50 z-50 justify-center items-center cursor-pointer`}
+      onClick={ev => {
+        if (!promptBoxRef.current?.contains(ev.target as Node)) {
+          closeEditor();
+        }
+      }}
+    >
+      <div
+        className="rounded-4 px-6 py-4 bg-white h-2/3 w-2/3 shadow cursor-default flex"
+        ref={promptBoxRef}
+      >
+        <div className="flex flex-col w-full">
+          <h3 className="text-lg font-semibold text-gray-700 mb-2">Prompt</h3>
+          <PromptInput
+            currentPrompt={currentPrompt}
+            setCurrentPrompt={setCurrentPrompt}
+            keepResults={keepResults}
+            promptResult={promptResult}
+            setPromptResult={setPromptResult}
+          />
+          {task === "classify" ? (
+            <>
+              <h3 className="text-lg font-semibold text-gray-700 mb-2 mt-3">Options</h3>
+              <ClassifyOptions options={classifyOptions} setOptions={setClassifyOptions} />
+            </>
+          ) : null}
+          {task === "classify" ? (
+            <ClassifyResults options={classifyOptions} results={classifyResults} />
+          ) : null}
+        </div>
+        <div className="flex flex-col w-48 pl-6">
+          <h3 className="text-lg font-semibold text-gray-700 mb-2">Agent</h3>
+          <label className="text-sm mb-1 text-gray-700">Model</label>
+          <Select onChange={ev => setSelectedAgent(ev.target.value)} value={selectedAgent}>
+            {agents.map(el => (
+              <option key={el} value={el}>
+                {el}
+              </option>
+            ))}
+          </Select>
+          <label className="text-sm mb-1 mt-3 text-gray-700">Task</label>
+          <Select onChange={ev => setTask(ev.target.value)} value={task}>
+            <option value="complete">Complete</option>
+            <option value="classify">Classify</option>
+          </Select>
+          <label className="text-sm mb-1 mt-3 text-gray-700">Stop sequences</label>
+          <label className="text-sm text-gray-500 mb-1">
+            Enter an option, then press &apos;Tab&apos; to add
+          </label>
+          <ClassifyOptions options={stop} setOptions={setStop} placeholder="" />
```

Comment:
It's confusing that <ClassifyOptions> is used here, can you name the component more generically?